e2e/tests/test_smoke.py

```diff
@@ -0,0 +1,146 @@
+import unittest
+import time
+import timeout_decorator
+import subprocess
+import warnings
+import docker
+from halo import Halo
+
+from kubernetes import client, config, utils
+
+
+class SmokeTestCase(unittest.TestCase):
+    '''
+    Test the most basic functions of the operator.
+    '''
+
+    TEST_TIMEOUT_SEC = 600
+    RETRY_TIMEOUT_SEC = 5
+
+    @classmethod
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def setUpClass(cls):
+        '''
+        Deploy operator to a "kind" cluster created by /e2e/run.sh using examples from /manifests.
+        This operator deployment is to be shared among all tests.
+
+        /e2e/run.sh deletes the 'kind' cluster after successful run along with all operator-related entities.
+        In the case of test failure the cluster will stay to enable manual examination;
+        next invocation of "make e2e" will re-create it.
+        '''
+
+        k8s_api = K8sApi()
+
+        # HACK
+        # 1. creating RBAC entites with a separate client fails
+        #    with "AttributeError: object has no attribute 'select_header_accept'"
+        # 2. utils.create_from_yaml cannot create multiple entites from a single file
+        subprocess.run(["kubectl", "create", "-f", "manifests/operator-service-account-rbac.yaml"])
+
+        for filename in ["configmap.yaml", "postgres-operator.yaml"]:
+            utils.create_from_yaml(k8s_api.k8s_client, "manifests/" + filename)
+
+        # submit the most recent operator image built locally
+        # HACK assumes "images list" returns the most recent image at index 0
+        docker_client = docker.from_env()
+        image = docker_client.images.list(name="registry.opensource.zalan.do/acid/postgres-operator")[0].tags[0]
+        body = {
+            "spec": {
+                "template": {
+                    "spec": {
+                        "containers": [
+                            {
+                              "name": "postgres-operator",
+                              "image": image
+                            }
+                        ]
+                    }
+                 }
+            }
+        }
+        k8s_api.apps_v1.patch_namespaced_deployment("postgres-operator", "default", body)
```

Comment:
Wouldn't it be better to create the deployment with latest image straight away instead of patching an older version? Now it's also sort of a backwards compatibility check, right? ;) Any reason not to the create [api] for deployments (https://github.com/kubernetes-client/python/blob/master/examples/deployment_examples.py). Can still be changed later, though.

Reply:
yes, it is evidently better to configure deployment completely before creating it. there however two ways to achieve this goal:
1. modify the deployment manifest. That approach requires shuffling the file around and that is exactly [what the `run_operator_locally.sh` does](https://github.com/zalando/postgres-operator/blob/master/run_operator_locally.sh#L128) I preferred standard k8s API to that approach.
2. upload the deployment definition in memory and modify it. That's the proper approach which we should use  once we see the tests we need actually fit into the proposed runner structure.

Reply:
Not following completely, but is python so you can also just have nice template strings using

```
deployment = """"
...
""""
```

And use e.g. jinja2 templates to resolve.